### PR TITLE
Adjustments for configure --enable-rexgen

### PR DESCRIPTION
**Pull request by frank-dittrich**
Adjust john to "recent" rexgen interface changes,
point out known issues and possibility that
rexgen support might have to be dropped in future releases

See https://github.com/openwall/john/issues/4640#issuecomment-803584145 for more details

### CODE REVIEW

File: doc/README.librexgen [hunk 1]
```diff
@@ -2,6 +2,12 @@ Rexgen.txt - This document describes how to use the rexgen library, to perform
 regex expression work within JtR.  Rexgen library is copywrite Jan Starke,
 jan.starke@outofbed.org   The regex.c code in JtR by JimF, Spring 2014.
 
+WARNING
+
+Since maintenance for the rexgen library appears to have stopped for almost
+two years, support for --regex might have to be dropped in future releases!
```
Comment (solardiz) [inline doc/README.librexgen]: Maybe mention specific calendar years, as the "two years" reference would be non-specific to someone reading this years later.

File: doc/README.librexgen [hunk 2]
```diff
@@ -82,22 +88,52 @@ https://github.com/teeshop/rexgen
...
+
+Checkout this latest known good commit:
+
+cd rexgen
+git checkout 5b2f4b159ec948c1f9429eca4389ca2adc9c0b07
+
+
+
+Rexgen requires a cmake environment, along with flex and bison
+to be installed and working.  To build, see instructions on this page:
```
Comment (solardiz) [inline doc/README.librexgen]: Maybe replace the first sentence with simply "Building rexgen requires cmake, flex, and bison."

File: doc/README.librexgen [hunk 3]
```diff
@@ -2,6 +2,12 @@ Rexgen.txt - This document describes how to use the rexgen library, to perform
 regex expression work within JtR.  Rexgen library is copywrite Jan Starke,
 jan.starke@outofbed.org   The regex.c code in JtR by JimF, Spring 2014.
```
Comment (solardiz) [inline doc/README.librexgen:3]: It's not currently called `Rexgen.txt`, and maybe `copyright`? Putting this together, with some other edits:

```
This document describes how to use the rexgen library to have JtR generate
candidate passwords matching a regular expression.   The rexgen library is by
Jan Starke.  The regex.c code in JtR is by JimF, Spring 2014.
```

**Review by solardiz — APPROVED** [review]
OK with me.  Feel free to make my suggested edits or not.

File: doc/README.librexgen (hunk 3 above)
Comment (frank-dittrich) [reply doc/README.librexgen:3]:
```
$ git blame regex.c|cut -b 13-26|sort|uniq -c|sort -nr
    252 JimF
    135 magnum  
     36 Frank Dittrich
     24 Jan Starke
$ git blame regex.c|grep Starke |head -n 1|cut -b 1-94
e40f1d3570 (JimF           2014-03-27 22:21:48 +0100   9)  * Author if this file is Jan Starke
```
IIRC, Jan Starke in 2014 provided the original Rexgen.txt which JimF renamed to README.librexgen, and he also provided the initial John the Ripper patch which JimF committed.
JimF among other things added support for --restore, magnum hybrid mode support, encoding changes, etc., Jan Starke later contributed adjustments to new libgexgen releases, I just contributed some minor fixes.

Comment (solardiz) [reply doc/README.librexgen:3]: OK, it's tricky. You want to leave this paragraph as-is for now?

Comment (frank-dittrich) [reply doc/README.librexgen:3]: Yes, I don't want to be guilty of making matters worse.

**Pull request merged**